pkgs/development/libraries/linenoise/default.nix

```diff
@@ -0,0 +1,29 @@
+  { stdenv, fetchgit }:
+
+  stdenv.mkDerivation rec {
+    name = "linenoise-${version}";
+    version = "git-2016-09-30";
+
+    src = fetchgit {
+      url = "https://github.com/antirez/linenoise.git";
+      rev = "c894b9e59f02203dbe4e2be657572cf88c4230c3";
+      sha256 = "0wasql7ph5g473zxhc2z47z3pjp42q0dsn4gpijwzbxawid71b4w";
+    };
+
+    phases = ["unpackPhase" "buildPhase" "installPhase"];
```

Comment:
Which phase is problematic? The default phases are supposed to be somewhat smart wrt. not doing things unless appropriate files are found in the build toplevel.

For some phases there are flags, like `dontBuild = true` or you can override the phase to a noop, as in `buildPhase = ":"`. See https://nixos.org/nixpkgs/manual/#sec-stdenv-phases for details.